src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -183,9 +199,52 @@ class StickyOutlineElement {
 		}
 		return new StickyOutlineElement(
 			range,
-			children
+			children,
+			undefined
 		);
 	}
+
+	public static fromFoldingModel(foldingModel: FoldingModel): StickyOutlineElement {
+		const regions = foldingModel.getRegions();
+		const startIndexes = regions.getStartIndexes();
+		const endIndexes = regions.getEndIndexes();
+		if (startIndexes.length === 0) {
+			return new StickyOutlineElement(
+				new StickyRange(-1, -1),
+				[],
+				undefined
+			);
+		}
+		let range = undefined;
+		const stackOfParents: StickyRange[] = [];
+
+		const stickyOutlineElement = new StickyOutlineElement(
+			undefined,
+			[],
+			undefined
+		);
+		let parentStickyOutlineElement = stickyOutlineElement;
+
+		for (let i = 0; i < startIndexes.length; i++) {
+			range = new StickyRange(startIndexes[i], endIndexes[i]);
+			while (stackOfParents.length !== 0 && (range.startLineNumber < (stackOfParents[stackOfParents.length - 1] as StickyRange).startLineNumber || range.endLineNumber > (stackOfParents[stackOfParents.length - 1] as StickyRange).endLineNumber)) {
```

Comment:
The `as StickyRange` cast aren't needed - I believe

Reply:
That's right, back when I added it, there was an error without this type cast. Now it's no longer necessary.